hopsworks-common/src/main/java/io/hops/hopsworks/common/serving/sklearn/LocalhostSkLearnServingController.java

```diff
@@ -0,0 +1,232 @@
+        .addCommand(Paths.get(serving.getArtifactPath()).toString())
+        .addCommand(String.valueOf(port))
+        .addCommand(secretDir.toString())
+        .addCommand(project.getName() + USER_NAME_DELIMITER + user.getUsername())
+        .addCommand(project.getName().toLowerCase())
+        .addCommand(settings.getAnacondaProjectDir(project) + "/bin/python")
+        .addCommand(certificateMaterializer.getUserTransientKeystorePath(project, user))
+        .addCommand(certificateMaterializer.getUserTransientTruststorePath(project, user))
+        .addCommand(certificateMaterializer.getUserTransientPasswordPath(project, user))
+        .addCommand(serving.getName())
+        .setWaitTimeout(2L, TimeUnit.MINUTES)
+        .ignoreOutErrStreams(true)
+        .build();
+    logger.log(Level.INFO, processDescriptor.toString());
+    try {
+      // Materialized TLS certificates so that user can read from HDFS inside python script
+      certificateMaterializer.materializeCertificatesLocal(user.getUsername(), project.getName());
+      ProcessResult processResult = osProcessExecutor.execute(processDescriptor);
+      if (processResult.getExitCode() != 0) {
+        // Startup process failed for some reason
+        serving.setLocalPid(PID_STOPPED);
+        servingFacade.updateDbObject(serving, project);
+        throw new ServingException(RESTCodes.ServingErrorCode.LIFECYCLEERRORINT, Level.INFO);
+      }
+
+      // Read the pid for TensorFlow Serving server
+      Path pidFilePath = Paths.get(secretDir.toString(), "sklearn_flask_server.pid");
+      String pidContents = Files.readFirstLine(pidFilePath.toFile(), Charset.defaultCharset());
+
+      // Update the info in the db
+      serving.setLocalPid(Integer.valueOf(pidContents));
+      serving.setLocalPort(port);
+      servingFacade.updateDbObject(serving, project);
+    } catch (Exception ex) {
+      // Startup process failed for some reason
+      serving.setLocalPid(PID_STOPPED);
+      servingFacade.updateDbObject(serving, project);
+
+      throw new ServingException(RESTCodes.ServingErrorCode.LIFECYCLEERRORINT, Level.SEVERE, null,
+          ex.getMessage(), ex);
+
+    } finally {
+      if (settings.getHopsRpcTls()) {
+        certificateMaterializer.removeCertificatesLocal(user.getUsername(), project.getName());
+      }
+      // release lock on the serving entry
+      servingFacade.releaseLock(project, serving.getId());
+    }
+  }
+  
+  /**
+   * Validates user data for creating or updating a SkLearn Serving Instance
+   *
+   * @param servingWrapper the user data
+   * @param project the project to create the serving for
+   * @throws ServingException if the python environment is not activated for the project
+   */
+  public void validateUserInput(ServingWrapper servingWrapper, Project project) throws ServingException {
+
+    // Check that serving name is provided
+    if (Strings.isNullOrEmpty(servingWrapper.getServing().getName())) {
```

Comment:
You are already checking that the name is not null in the `LocalhostServingController`

Reply:
Done